src/support.cpp

```diff
@@ -1399,7 +1418,30 @@ void AreaSupport::generateSupportBottom(SliceDataStorage& storage, const SliceMe
         }
         Polygons bottoms;
         generateSupportInterfaceLayer(global_support_areas_per_layer[layer_idx], mesh_outlines, bottom_line_width, bottom_outline_offset, minimum_bottom_area, bottoms);
-        support_layers[layer_idx].support_bottom.add(bottoms);
+
+        if (layer_idx < support_layers.size() - 1)
+        {
+            Polygons support_bottoms;
+            const coord_t actual_support_offset = getActualSupportOffset();
+            const std::vector<PolygonsPart> bottom_parts = bottoms.splitIntoParts();
+            for (const PolygonsPart& bottom_part : bottom_parts)
+            {
+                // check if:
+                // - any interface polygons are printed above;
+                // - any support polygons are printed above, because some support could be pushed away by large XY distance or support could be so tiny that it skipped by offset due to support walls.
+                const bool is_any_support_present = !bottom_part.intersection(support_layers[layer_idx + 1].support_bottom).empty()
+                        || !bottom_part.intersection(global_support_areas_per_layer[layer_idx + 1].offset(actual_support_offset).offset(-actual_support_offset)).empty();
+                if (is_any_support_present)
+                {
+                    support_bottoms.add(bottom_part);
+                }
+            }
+            support_layers[layer_idx].support_bottom.add(support_bottoms);
+        }
+        else
+        {
+            support_layers[layer_idx].support_bottom.add(bottoms);
+        }
```

Comment:
This part should be moved into the `generateSupportInterfaceLayer` function or a separate one beneath it. That would also prevent the duplication here. The only changes really are the `+ 1` vs. `- 1`.

Reply:
Refactored, thank you